### DISCUSSION TITLE
ETAPI Saving Notes Quirks

### DISCUSSION THREAD

**Discussion by itsallyours** [TriliumNext/Trilium#3509] [Q&A] [answered]
I have been using the ETAPI for fixing some broken links inside of imports. 

Basic process

1. I grab the existing note content via the ETAPI
2. update it with the correct links (using internal href and data-note-path).  
3. PUT the updated content via the ETAPI

I have observed two things

1. Backlinks are never created by the ETAPI (easily fixed with a slight edit in the UI:  e.g. adding a space)
2. Revisions are never created by the ETAPI

Any simple adjustments that I can do via the ETAPI to address either of these.

(BTW awesome work)

**Comment by zadam** [comment, marked as answer]
Hi, thanks for reporting these issues:

> Backlinks are never created by the ETAPI (easily fixed with a slight edit in the UI: e.g. adding a space)

This is an omission in the update-content method (they get created in the create-note). Fixed.

> Revisions are never created by the ETAPI

This is intentional, the auto-creation of note revisions is IMHO suited only for UI. I created a separate method `POST /notes/{noteId}/note-revision` to do that.

**Reply by itsallyours** [reply]
Both changes work great!  Thank you!
